lib/util/eyeglass_modules.js

```diff
@@ -0,0 +1,283 @@
+  var branches = findBranchesByPath({
+    dependencies: this.tree
+  }, origin);
+  return branches.some(function(branch) {
+    if (branch.name === name || branch.dependencies[name]) {
+      return true;
+    }
+  });
+}
+
+function flattenModules(branch, registry) {
+  if (branch) {
+    // if the branch itself is a module, add it...
+    if (branch.isEyeglassModule) {
+      registry = registry || {};
+      registry[branch.name] = registry[branch.name] || [];
+      registry[branch.name].push(branch);
+    }
+
+    var dependencies = branch.dependencies;
+
+    if (dependencies) {
+      var result = Object.keys(dependencies).reduce(function(registry, name) {
+        return flattenModules(dependencies[name], registry);
+      }, registry || {});
+      return result;
+    }
+  }
+
+  return registry;
+}
+
+function checkVersions(versions, latest) {
+  return versions.map(function(mod) {
+    if (!semver.satisfies(mod.version, "^" + latest.version)) {
+      return {
+        name: mod.name,
+        left: mod,
+        right: latest
+      };
+    }
+  });
+}
+
+function dedupeModules(modules) {
+  if (!modules) {
+    return [];
+  }
+  var errors = [];
+  return Object.keys(modules).reduce(function(deduped, name) {
+    var versions = modules[name].sort(function(a, b) {
+      return semver.lt(a.version, b.version);
+    });
+    deduped[name] = versions.shift();
+    errors.push.apply(errors, checkVersions(versions, deduped[name]));
+    return deduped;
+  }, {});
+}
+
+function getModuleName(pkg) {
+  var name = normalizeEyeglassOptions(pkg.data.eyeglass).name || pkg.data.name;
+  if (!name) {
+    throw new Error(tick(pkg.path) + " does not contain a valid module name");
+  }
+  return name;
+}
+
+function normalizeEyeglassOptions(options) {
+  if (typeof options === "string") {
+    return {
+      exports: options
+    };
+  }
+  return typeof options === "object" ? options : {};
+}
+
+function getFinalModule(name) {
+  return this.collection[name];
+}
+
+function getExportsFileFromOptions(options) {
+  return normalizeEyeglassOptions(options).exports;
+}
+
+function getModuleExports(pkg, modulePath) {
+  var exportsFile = getExportsFileFromOptions(pkg.eyeglass) || pkg.main;
+  return path.join(modulePath, exportsFile || "");
+}
+
+function isEyeglassModule(pkg) {
+  return !!(pkg && includes(pkg.keywords, EYEGLASS_KEYWORD));
+}
+
+function resolveModule(pkg, isRoot) {
+  pkg = packageUtils.getPackage(pkg);
+  var isEyeglassMod = isEyeglassModule(pkg.data);
+  if (isEyeglassMod || (pkg.data && isRoot)) {
+    var modulePath = path.dirname(pkg.path);
+    if (typeof pkg.data.version === "object") {
+      console.log("pkg.data.version", pkg.data.version);
+    }
+    return {
+      name: getModuleName(pkg) || (isRoot && ROOT_ID),
+      rawName: pkg.data.name,
+      path: modulePath,
+      version: pkg.data.version,
+      main: getModuleExports(pkg.data, modulePath),
+      isEyeglassModule: isEyeglassMod,
```

Comment:
If the project root itself is not an eyeglass module. That's the only case.

This is important so that we can correctly preserve the dependency tree for checking access permission.